controllers/apps/component_utils.go

```diff
@@ -31,11 +31,6 @@ import (
 
 func delayUpdateInstanceSetSystemFields(obj workloads.InstanceSetSpec, pobj *workloads.InstanceSetSpec) {
```

Comment:
Remove delayUpdateInstanceSetSystemFields.

Reply:
I'm wondering if `delayUpdateKubeBlocksToolsImage` is still needed. Instancest supports in-place update for container images and will not cause pod restart.

Reply:
Is it available on all possible k8s versions?

Reply:
In place update of container images can be found up to [1.20's doc](https://github.com/kubernetes/website/blob/release-1.20/content/en/docs/concepts/workloads/pods/_index.md#pod-update-and-replacement). I didn't check older versions.

Reply:
We can submit a separate PR to remove the feature for delaying the update of the tools image.

Reply:
https://github.com/apecloud/kubeblocks/pull/8595